pkg/handler3/handler.go

```diff
@@ -92,94 +136,92 @@ func constructServerRelativeURL(gvString, etag string) string {
 }
 
 // NewOpenAPIService builds an OpenAPIService starting with the given spec.
-func NewOpenAPIService(spec *spec.Swagger) (*OpenAPIService, error) {
+func NewOpenAPIService() *OpenAPIService {
```

Comment:
As much as I want to replace this too, do we need to worry about backwards compatibility at all?

Reply:
Good question, there's basically 2 things that I've had to change in k/k because of this PR, basically:
```
commit 094d2126c80978dbf1a3963828f7e0790d0dc209
Author: Antoine Pelisse <apelisse@gmail.com>
Date:   Thu Dec 29 12:12:33 2022 -0800

    Update kubernetes code for minor API changes to kube-openapi

diff --git a/staging/src/k8s.io/apiserver/pkg/server/routes/openapi.go b/staging/src/k8s.io/apiserver/pkg/server/routes/openapi.go
index e6459ce3ea2..19dbbb03443 100644
--- a/staging/src/k8s.io/apiserver/pkg/server/routes/openapi.go
+++ b/staging/src/k8s.io/apiserver/pkg/server/routes/openapi.go
@@ -56,12 +56,8 @@ func (oa OpenAPI) InstallV2(c *restful.Container, mux *mux.PathRecorderMux) (*ha
 
 // InstallV3 adds the static group/versions defined in the RegisteredWebServices to the OpenAPI v3 spec
 func (oa OpenAPI) InstallV3(c *restful.Container, mux *mux.PathRecorderMux) *handler3.OpenAPIService {
-   openAPIVersionedService, err := handler3.NewOpenAPIService(nil)
-   if err != nil {
-       klog.Fatalf("Failed to create OpenAPIService: %v", err)
-   }
-
-   err = openAPIVersionedService.RegisterOpenAPIV3VersionedService("/openapi/v3", mux)
+   openAPIVersionedService := handler3.NewOpenAPIService()
+   err := openAPIVersionedService.RegisterOpenAPIV3VersionedService("/openapi/v3", mux)
    if err != nil {
        klog.Fatalf("Failed to register versioned open api spec for root: %v", err)
    }
diff --git a/staging/src/k8s.io/kube-aggregator/pkg/controllers/openapiv3/aggregator/aggregator.go b/staging/src/k8s.io/kube-aggregator/pkg/controllers/openapiv3/aggregator/aggregator.go
index 67e793077e9..8f4edffc595 100644
--- a/staging/src/k8s.io/kube-aggregator/pkg/controllers/openapiv3/aggregator/aggregator.go
+++ b/staging/src/k8s.io/kube-aggregator/pkg/controllers/openapiv3/aggregator/aggregator.go
@@ -93,10 +93,7 @@ func BuildAndRegisterAggregator(downloader Downloader, delegationTarget server.D
        i++
    }
 
-   handler, err := handler3.NewOpenAPIService(nil)
-   if err != nil {
-       return s, err
-   }
+   handler := handler3.NewOpenAPIService()
    s.openAPIV2ConverterHandler = handler
    openAPIV2ConverterMux := mux.NewPathRecorderMux(openAPIV2Converter)
    s.openAPIV2ConverterHandler.RegisterOpenAPIV3VersionedService("/openapi/v3", openAPIV2ConverterMux)
```
It's really nothing, but yeah, do we want to keep things strictly compatible, good question ...

Reply:
If we want to backport a bug fix in kube-openapi to earlier k8s versions, will they pick up this caching feature as a byproduct?

Reply:
Yeah, we would either have to branch kube-openapi which we've done a few times, or they would have to make these two minor changes 🤷

Reply:
How do you feel about that?